test/boost/disk_space_monitor_test.cc

```diff
@@ -0,0 +1,132 @@
+    auto stop_dsm = deferred_stop(dsm);
+
+    testlog.debug("normal_polling_interval={} high_polling_interval={} polling_interval_threshold={:.1f}",
+        dsm_config.normal_polling_interval.get(), dsm_config.high_polling_interval.get(), dsm_config.polling_interval_threshold.get());
+
+    auto started_at = manual_clock::now();
+    struct call {
+        manual_clock::time_point t;
+        float util;
+    };
+    std::vector<call> calls;
+    auto sub = dsm.listen([&] (const utils::disk_space_monitor<manual_clock>& dsm) -> future<> {
+        testlog.info("total_space={} free_space={} utilization={:.1f}%",
+            dsm.total_space(), dsm.free_space(), dsm.disk_utilization() * 100);
+        calls.emplace_back(manual_clock::now(), dsm.disk_utilization());
+        return make_ready_future();
+    });
+
+    struct seconds {
+        long millis;
+
+        seconds(manual_clock::duration d) : millis(duration_cast<std::chrono::milliseconds>(d).count()) {}
+        seconds(manual_clock::time_point t) : millis(duration_cast<std::chrono::milliseconds>(t.time_since_epoch()).count()) {}
+
+        double count() const noexcept { return (double)millis / 1000; }
+    };
+
+    auto wait_for_call = [&] {
+        auto started_waiting_at = manual_clock::now();
+        auto start_size = calls.size();
+        while (calls.size() == start_size && seconds(manual_clock::now() - started_waiting_at).count() <= dsm_config.normal_polling_interval.get() * 2) {
+            testlog.debug("{:.3f}: advancing clock", seconds(manual_clock::now() - started_at).count());
+            manual_clock::advance(100ms);
+            yield().get();
+        }
+        return calls.size();
+    };
+
+    auto require_in_range = [] (const manual_clock::time_point& t, const manual_clock::time_point& t0, const utils::updateable_value<int>& interval) {
+        auto delta = seconds(t - t0).count();
+        auto expected = interval.get();
+        BOOST_REQUIRE_GE(delta, expected);
+        BOOST_REQUIRE_LE(delta, expected + 1);
+    };
+
+    BOOST_REQUIRE_EQUAL(wait_for_call(), 1);
+    require_in_range(calls[0].t, started_at, dsm_config.normal_polling_interval);
+    BOOST_REQUIRE_EQUAL(calls[0].util, 0.);
+
+    dsm.set_utilization(dsm_config.polling_interval_threshold * .5);
+    BOOST_REQUIRE_EQUAL(wait_for_call(), 2);
+    require_in_range(calls[1].t, calls[0].t, dsm_config.normal_polling_interval);
+    BOOST_REQUIRE_EQUAL(calls[1].util, dsm_config.polling_interval_threshold * .5);
+
+    dsm.set_utilization(dsm_config.polling_interval_threshold);
+    BOOST_REQUIRE_EQUAL(wait_for_call(), 3);
+    require_in_range(calls[2].t, calls[1].t, dsm_config.normal_polling_interval);
+    BOOST_REQUIRE_EQUAL(calls[2].util, dsm_config.polling_interval_threshold);
+
+    BOOST_REQUIRE_EQUAL(wait_for_call(), 4);
+    require_in_range(calls[3].t, calls[2].t, dsm_config.high_polling_interval);
+    BOOST_REQUIRE_EQUAL(calls[3].util, dsm_config.polling_interval_threshold);
```

Comment:
What are we testing here?

Reply:
The monitor's polling interval as a function of (simulated) disk utilization.

Not the measurement of thereof.

Reply:
Well, we should test the actual measurement.

Otherwise, "return 0" would pass the tests too.

Reply:
The problem I saw with this is that if we use a live filesystem, e.g. /tmp then its utilization might change while we measure it, and if we want a clean filesystem, say using a loopback mount, we'll need to be sudoers to mount this filesystem for the test.  If that's ok, then it's doable.

Reply:
Edit: I found `udisksctl` that seems to allow setting up and mounting a loop device in user mode.
Will send a version using it.
It should be added to the frozen toolchain

Reply:
Cool! Thanks for pointing that out.
I'll try to follow its steps then.

Reply:
Done in v10 ([9a3db9a](https://github.com/scylladb/scylladb/commit/9a3db9ac9e444e6ef85f52f0223d11cf6d89b4e2))